pinot-core/src/test/java/org/apache/pinot/queries/NativeAndLuceneComparisonTest.java

```diff
@@ -0,0 +1,295 @@
+    return ImmutableSegmentLoader.load(new File(INDEX_DIR, SEGMENT_NAME_NATIVE), indexLoadingConfig);
+  }
+
+  private void testSelectionResults(String query, int expectedResultSize, List<Serializable[]> expectedResults) {
+    Operator<IntermediateResultsBlock> operator = getOperatorForSqlQuery(query);
+    IntermediateResultsBlock operatorResult = operator.nextBlock();
+    List<Object[]> resultset = (List<Object[]>) operatorResult.getSelectionResult();
+    Assert.assertNotNull(resultset);
+    Assert.assertEquals(resultset.size(), expectedResultSize);
+    if (expectedResults != null) {
+      for (int i = 0; i < expectedResultSize; i++) {
+        Object[] actualRow = resultset.get(i);
+        Object[] expectedRow = expectedResults.get(i);
+        Assert.assertEquals(actualRow.length, expectedRow.length);
+        for (int j = 0; j < actualRow.length; j++) {
+          Object actualColValue = actualRow[j];
+          Object expectedColValue = expectedRow[j];
+          Assert.assertEquals(actualColValue, expectedColValue);
+        }
+      }
+    }
+  }
+
+  @Test
+  public void testQueries() {
+    String query = "SELECT * FROM MyTable WHERE TEXT_MATCH(DOMAIN_NAMES_LUCENE, 'www.domain1%') LIMIT 50000";
+    _indexSegment = _luceneSegment;
+    _indexSegments = Arrays.asList(_indexSegment);
+    testSelectionResults(query, 256, null);
+
+    _indexSegment = _nativeIndexSegment;
+    _indexSegments = Arrays.asList(_nativeIndexSegment);
+    query = "SELECT * FROM MyTable WHERE DOMAIN_NAMES_NATIVE CONTAINS 'www.domain1.*' LIMIT 50000";
+    testSelectionResults(query, 256, null);
+
+    query = "SELECT * FROM MyTable WHERE TEXT_MATCH(DOMAIN_NAMES_LUCENE, '%www.domain1') LIMIT 50000";
+    _indexSegment = _luceneSegment;
+    _indexSegments = Arrays.asList(_indexSegment);
+    testSelectionResults(query, 256, null);
+
+    query = "SELECT * FROM MyTable WHERE DOMAIN_NAMES_NATIVE CONTAINS '.*www.domain1' LIMIT 50000";
+    _indexSegment = _nativeIndexSegment;
+    _indexSegments = Arrays.asList(_nativeIndexSegment);
+    testSelectionResults(query, 256, null);
+
+    query = "SELECT * FROM MyTable WHERE TEXT_MATCH(DOMAIN_NAMES_LUCENE, '%www.domain1%') LIMIT 50000";
+    _indexSegment = _luceneSegment;
+    _indexSegments = Arrays.asList(_indexSegment);
+    testSelectionResults(query, 256, null);
+
+    query = "SELECT * FROM MyTable WHERE DOMAIN_NAMES_NATIVE CONTAINS '.*www.domain1.*' LIMIT 50000";
+    _indexSegment = _nativeIndexSegment;
+    _indexSegments = Arrays.asList(_nativeIndexSegment);
+    testSelectionResults(query, 256, null);
+
+    query = "SELECT * FROM MyTable WHERE TEXT_MATCH(DOMAIN_NAMES_LUCENE, 'www.domain1% AND %www.domain1') LIMIT 50000";
+    _indexSegment = _luceneSegment;
+    _indexSegments = Arrays.asList(_indexSegment);
+    testSelectionResults(query, 256, null);
+
+    query =
+        "SELECT * FROM MyTable WHERE DOMAIN_NAMES_NATIVE CONTAINS 'www.domain1.*' AND DOMAIN_NAMES_NATIVE CONTAINS '"
```

Comment:
We should be able to support `SELECT * FROM MyTable WHERE DOMAIN_NAMES_NATIVE CONTAINS 'www.domain1% AND %www.domain1'`

Reply:
As discussed, this is a follow up item